### ISSUE TITLE
Unable to use go-routine to make request to AWS bucket

### ISSUE THREAD

**Issue by ConradKurth** [golang/go#13707]
So I found this issue when switching to 1.6 from 1.5.2. The issue is when i try to get some data from my AWS bucket. I am using github.com/mitchellh/goamz/aws and github.com/mitchellh/goamz/s3 to get the data. What happens is that if i call the method "Get"  in a go-routine the request will not go through, but if I do not put the call in a go-routine it will work just fine.

**Comment by ianlancetaylor** [comment]
Please show us a small test case demonstrating the problem.

**Comment by ConradKurth** [comment]
hey! thanks for getting back to me. After I looked at it more I found the underlying issue. I had an infinite loop later in my main function which i used for debugging purposes. Anyways, when main goroutine would reach that for loop it wouldn't release control so another routine would run. I thought I read somewhere in Go 1.2 ( I believe) this was fixed, but maybe not in the way I thought. Which is more strange is that for some reason the GOMAXPROCS is being defaulted to just 1 when i am running either 1.6 or 1.5.2 ( i have an i7). When I set GOMAXPROCS to higher than 1 it will work. Thoughts?
